x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1627,9 +1627,6 @@ public Function<String, Predicate<String>> getFieldFilter() {
                 if (indexPermissions == null) {
                     return MapperPlugin.NOOP_FIELD_PREDICATE;
                 }
-                if (indexPermissions.isGranted() == false) {
-                    throw new IllegalStateException("unexpected call to getFieldFilter for index [" + index + "] which is not granted");
-                }
```

Comment:
This is the only actual place where we have an explicit safeguard check for `indexPermissions.isGranted() == false`, and where replacing with a `null` is not equivalent.

Personally I'm not very worried about removing this safeguard. But if you are, we can discuss to put other equivalent safeguards in place.